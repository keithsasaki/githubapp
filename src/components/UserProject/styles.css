#project-name{
    color: var(--color-purple);
    font-size: 3.6rem;
    font-weight: normal;
}

#project-description {
    font-size: 2.1rem;
    font-weight: 300;
    padding-bottom: 1rem;
}   

#user-project {
    display: flex;
    flex-direction: column;
}

#result {
    display: flex;
    flex-direction: row
}

#user-informations{
    margin-right: 7rem;
}