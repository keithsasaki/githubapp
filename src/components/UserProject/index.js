import React from 'react';
import IconAndText from '../IconAndText';
import StarIcon from '../../assets/icons/star.svg'

import './styles.css';

function UserProject(props) {
    return (
        <div id="user-project">
            <span id="project-name">{props.repo.name}</span>
            <span id="project-description">{props.repo.description}</span>
            <IconAndText icon={StarIcon} text={props.repo.stargazers_count} />
        </div>
    );
}

export default UserProject;