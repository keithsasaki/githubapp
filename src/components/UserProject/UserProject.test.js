import React from 'react';
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';
import UserProject from '.';

describe("Testing UserProject Component", () => {
    const repositorie = {
        projectName:"Projeto github app",
        projectDescription:"Projeto em react para consumir api do Github",
        stargazers_count: 30
    }
    it("matches snapshot", () => {
        const tree = renderer.create(<UserProject repo={repositorie}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})