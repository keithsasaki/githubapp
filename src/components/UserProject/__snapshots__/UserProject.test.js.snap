// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserProject Component matches snapshot 1`] = `
<div
  id="user-project"
>
  <span
    id="project-name"
  />
  <span
    id="project-description"
  />
  <div
    className="icon-and-text"
  >
    <img
      src="star.svg"
    />
    <span>
       
      30
       
    </span>
  </div>
</div>
`;
